pkg/types/azure/validation/platform.go

```diff
@@ -40,4 +43,16 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 		}
 	}
 	return allErrs
+
+}
+
+// ValidateClusterName confirms that the provided cluster name matches GCP naming requirements.
+func ValidateClusterName(clusterName string) error {
+	azureResourceFmt := `^[a-z][a-z0-9-]{1,61}[a-z0-9]$`
+
+	re := regexp.MustCompile("^" + azureResourceFmt + "$")
```

Comment:
`azureResourceFmt` already contains the `^` and `$` regex tokens (these tokens require that no characters come before or after respectively; so they indicate that the regex matches the whole string), so they should not be added again here.